src/ui/public/visualize/visualize.js

```diff
@@ -52,6 +52,13 @@ uiModules
         });
 
         $scope.vis = $scope.savedObj.vis;
+        $scope.vis.createInheritedSearchSource = () => {
```

Comment:
This should not live inside visualize. If we really need something like this I would put it on vis.API.

I am wondering if we can avoid this completely somehow ?
- could map visualization request a new request to be executed and receive new data and rerender based on that ?
( so map would not be doing elasticsearch requests at all )

also currently this won't work at all if someone included visualization with <visualization> directive instead of <visualize>

Reply:
@timroes any ideas ?

Reply:
@ppisljar wrt. to 

> could map visualization request a new request to be executed and receive new data and rerender based on that ?
( so map would not be doing elasticsearch requests at all )

Not sure if this would work in this case. The fit-bounds operation needs a request for additional metadata, but this is separate from the data on screen that is retrieved by the request-handler.